src/ext.rs

```diff
@@ -238,3 +245,15 @@ pub fn address() -> Address {
 pub fn log(topics: &[H256], data: &[u8]) {
 	unsafe { external::elog(topics.as_ptr() as *const u8, topics.len() as u32, data.as_ptr(), data.len() as u32); }
 }
+
+/// Allocates and requests call arguments (input) from the runtime
+pub fn input() -> pwasm_std::Vec<u8> {
+	let mut data = pwasm_std::Vec::with_capacity(unsafe { external::input_length() } as usize);
```

Comment:
I think it would be better if we make `input_length == 0` special case and return Vec::new early